src/main/predict.cpp

```diff
@@ -204,8 +204,22 @@ void go(Config config) {
     }
     if(verbose) cout << "inputFile: '" << config.inputFile << "'"<< endl;
     if(config.inputFile == "") {
-        cin.read(reinterpret_cast< char * >(inputData), inputCubeSize * config.batchSize * 4l);
-        more = !cin.eof();
+	const int size = inputCubeSize * config.batchSize * 4l;
+	char *bytes = new char[size];
+
+	cin.read(bytes, size);
+	
+	float *bytes_as_float_pointer = (float*)bytes;
+
+	for (int i = 0; i < size / 4, i++;)
```

Comment:
this division will run each iteration of the loop.  its going to be quite fast anyway, but if you move it outside of the loop, then it is only calculated once.